services/auth.service.js

```diff
@@ -1,17 +1,43 @@
 import { EntityService } from './entity.service.js';
-
+import bcrypt from 'bcrypt';
 export class AuthService {
   constructor(db) {
     this.service = new EntityService(db, 'user');
   }
 
   async logIn(email, password) {
-    const user = await this.service.getOne({ email, password });
+    const user = await this.service.getOne({ email });
 
     if (!user) {
       throw new Error('Wrong credentials');
     }
 
+    const checkPassword = await bcrypt.compare(password, user.password);
+
+    if (!checkPassword) {
+      throw new Error(`Wrong credentials`);
+    }
+
     return user;
   }
+
+  async register(email, password, username) {
+    const user = await this.service.getOneConditional({ email, username });
+
+    if (user) {
+      if (user.email !== email && user.username === username) {
+        throw new Error(
+          `The user with username ${username} is already registered`
+        );
+      } else {
+        throw new Error(`User with email ${email} is already registered`);
+      }
+    }
+
+    const hashPassword = await bcrypt.hash(password, 3);
+
+    this.service.create({ email, password: hashPassword, username });
+
+    return this.service.getOneConditional({ email });
```

Comment:
Why are you using getOneConditional?
Try smth like:

return this.service.create...